### DISCUSSION TITLE
Let's agree on a date for TAPI-signaling WIP-0017!

### DISCUSSION THREAD

**Discussion by aesedepece** [witnet/witnet-rust#2055] [Q&A]
[WIP-0017 (Add median to RADON reducers)](https://github.com/witnet/WIPs/blob/master/wip-0017.md) seems beneficial to the protocol and is apparently facing no objections so far.

Let's agree on the starting date of its signaling period!

**Comment by lrubiorod** [comment]
Mmm... I propose 21st September 9.00 UTC. In 2 weeks, 1 for complete it and merge, and other week for people to updating. But if we want to add also RNG activation, maybe we should wait a little more?

**Reply by aesedepece** [reply]
Ok, then let's wait for the RNG WIP.

As per [WIP0014§(20)4](https://github.com/witnet/WIPs/blob/master/wip-0014.md#changes-to-wip-process), if both WIPs are in the Draft stage, their authors can propose joint activation.

**Reply by drcpu-github** [reply]
I don't think either update will be contentious, but I think the precedent of node operators not being able to choose if they want to support one upgrade and not the other is particularly interesting.

**Reply by aesedepece** [reply]
I guess so. What do you mean by "not being able to choose"? Not finding it easy to make a decision? Or not given the chance to opt-out?

**Reply by drcpu-github** [reply]
The latter, not given the chance to opt-out of one upgrade while still accepting the other. Technically, WIP0014 and WIP0016 already created a precedent though.

I'm really doing some major nitpicking here, but I just want to voice that concern.

**Reply by aesedepece** [reply]
I totally get your point.

Adding a new reducer and adding RNG functionality seem two different features with very different implications, and I can perfectly imagine the case in which one node operator may be supportive of one but not the other.

Something inside of me sort is sort of begging for separate signaling.

However, let's keep in mind that TAPI is not a voting / governance mechanism. If anyone in the community has strong objections against some improvement proposal, there's time and space for voicing those concerns before the improvement is implemented or signaled.

Even if witnet-rust provides the "oppose this improvement" feature, I tend to think of it as a last resort thing and it shouldn't be used for objection but rather for calling out the witnet-rust developers in the case they included some contentious protocol change that lacks the favor of the rest of the community.

**Reply by drcpu-github** [reply]
I guess that is true. If a WIP is truly contentious, there are plenty of channels to discuss this first and bundling activation of previously discussed non-contentious WIP's is not too bad in that sense.
